valhalla/baldr/edgeinfo.h

```diff
@@ -147,10 +148,19 @@ class EdgeInfo {
 
   /**
    * Convenience method to get the names for an edge
-   * @return   Returns a list (vector) of names.
+   * @return   Returns a list (vector) of untagged names.
    */
   std::vector<std::string> GetNames() const;
 
+  /** get the name & is_tagged flag
+   *
+   *  This one does not calculate the types but has the is_tagged flag
+   *  Like GetNamesAndTypes but without using memory for the types
+   *
+   *  @return iReturns a list (vector) (name, is_tagged) pairs
```

Comment:
are the docstrings correct? to me the bools that come back in both cases are to indicate whether or not the name is a route number, not just if its generically tagged. and then the thing that it returns as the third piece in the tuple below isnt the route number itself but rather the tag type, ie `TaggedValue`

Reply:
I dont know, if the docstrings are correct,
To me this does not make sense,
The vector has name on the std::string and route number on the bool?
```
* @return   Returns a list (vector) of name/route number pairs.
*/
 std::vector<std::pair<std::string, bool>> GetNamesAndTypes(std::vector<uint8_t>& types,
                                                             bool include_tagged_names = false) const;
```

Reply:
For me, tags come from osm: like: highway=motorway, boundary=yes  etc.
So I am very very confused:

Here is about the function I based the code on: (The new function I made was based on this one)
https://github.com/valhalla/valhalla/blob/master/valhalla/baldr/edgeinfo.h#L162

From the docstrings:

> Returns a list (vector) of name/route number pairs.

which is std::vector<std::pair<std::string, bool>>

Therefore the bool value is a route number.

Which is not, it is just an indicator that:
for the ith value on the returned vector when the bool value is true then the value on the ith element on the `types` vector is valid.

But then you mention:

> so the boolean  to me the bools that come back in both cases are to indicate whether or not the name is a route number, 

what do you mean?
like [route 66](https://en.wikipedia.org/wiki/U.S._Route_66)? 
Then the name is '66' and that name will be returned only when the function is called with `true`

Maybe you mean:
so the boolean  to me the bools that come back in both cases are to indicate whether or not the value in `type` is a route number

Definitively the function that uses this one does not use the `types` (aka route number)

--------

In the new function I made, there is no `types` (wich I think is your route number)
So I am just keeping the indicator that the name has a `type`.

```
@return Returns a list (vector) (name, is_tagged) pairs 
```
Maybe you want
```
@return Returns a list (vector) (name, has_tag) pairs 
```

Reply:
@kevinkreiser  Need feedback here

Reply:
ok so we store a bunch of variable sized stuff in `edgeinfo` that makes it into the text list. some of those things are like what you'd expect, such as `main street` but some of the records are also marked with a boolean flag, roads like `I-81` or as you mentioned `US-66`. for the latter 2 that flag tells us that its not just a regular road name, its a shielded road. shielded roads get special treatment in the forming of directions and narrative (the textual instructions). if you have a look over at the comment in edgeinfo.h it tells you basically what im trying to explain here: https://github.com/valhalla/valhalla/blob/15dedaac1b26730d78bc1f4ae988ad88f4c95264/valhalla/baldr/edgeinfo.h#L37-L38

so yeah the boolean in both cases is `is_route_num` ie whether or not the name is the route relations name.

now we also have this concept of "tagged" values. this is not like osm exactly. we have space for an enumeration (1 byte worth), and then can attach with that 1 byte a bunch of data, whatever it may be in the same text record where we store normal textual stuff.

the code looks at the single bit (is_tagged) on the nameinfo object and if that is the case it says ok the first byte of the bytes in that record is the value of the enum for the tag. in this case the enum is here: https://github.com/valhalla/valhalla/blob/master/valhalla/baldr/graphconstants.h#L378

so its not completely free form like OSM but rather we pick and choose which things we want ot support, give them a number in the enum, that becomes the first byte of the text record and then depending on which enum it was we would have different run time parsing of the bytes that followed.

this is a convenient way for us to randomly add non performance critical metadata to the dataset. especially data that is sparse (meaning it doesnt occur on all edge in the graph)

hopefully that explains what you are seeing here in edgeinfo :smile: